File: solutions/lua/hamming/1/hamming.lua
local Hamming = {}

function Hamming.compute(a,b)

    local n = string.len(a)
    if n ~= string.len(b) then
        return -1
    end
    
    local distance = 0
    for i = 1, n do
        if string.byte(a, i) ~= string.byte(b, i) then
            distance = distance + 1
        end
    end
    return distance
end

return Hamming

File: solutions/javascript/hamming/1/hamming.js
// Computes the Hamming distance between two strands of DNA.
// The distance is the number of differences in the strands.
// If the strands have a different length, an error is returned.
export function compute(dna1, dna2) {
  
  if (dna1.length != dna2.length) throw new Error('strands must be of equal length');

  let distance = 0;
  for (let i = 0; i < dna1.length; i++) {
    if (dna1.charAt(i) !== dna2.charAt(i)) distance++;
  }
  return distance;
}
